File: backend/src/Shalom.Application/Contracts/ImportantDateDto.cs
namespace Shalom.Application.Contracts;

/// <summary>
/// A recurring date with its derived next occurrence. <paramref name="NextOccurrence"/>
/// and <paramref name="DaysUntil"/> are computed from the server's local today
/// (Feb 29 birthdays land on Feb 28 in non-leap years).
/// </summary>
public record ImportantDateDto(
    Guid Id,
    Guid PersonId,
    string Label,
    int Month,
    int Day,
    int? Year,
    int LeadDays,
    DateOnly NextOccurrence,
    int DaysUntil);

File: frontend/projects/api/src/lib/models/important-date-dto.ts
/**
 * A recurring date with its server-derived next occurrence (`yyyy-MM-dd`)
 * and `daysUntil` from the server's local today (ADR-004). Feb 29 lands on
 * Feb 28 in non-leap years.
 */
export interface ImportantDateDto {
  readonly id: string;
  readonly personId: string;
  readonly label: string;
  readonly month: number;
  readonly day: number;
  readonly year: number | null;
  readonly leadDays: number;
  readonly nextOccurrence: string;
  readonly daysUntil: number;
}
